qiskit/circuit/quantumregister.py

```diff
@@ -30,3 +32,37 @@ class QuantumRegister(Register):
     def qasm(self):
         """Return OPENQASM string for this register."""
         return "qreg %s[%d];" % (self.name, self.size)
+
+    def __getitem__(self, key):
+        """
+        Arg:
+            key (int|slice|list): index of the qubit to be retrieved.
+
+        Returns:
+            tuple[Register, int]: a tuple in the form `(self, key)` if key is int.
+                If key is a slice, return a `list((self,key))`.
+
+        Raises:
+            QiskitError: if the `key` is not an integer.
+            QiskitIndexError: if the `key` is not in the range
+                `(0, self.size)`.
+        """
+        return self.getitem(QuBit, key)
+
+    def __iter__(self):
+        """
+        Yields:
+            Qubit: an iterator over the qubits in the register.
+        """
+        for bit in range(self.size):
```

Comment:
`index` might be more accurate here (`bit` isn't {0,1} nor a `Bit` nor a `QuBit`)

Reply:
fixed in 05234367